.github/workflows/pre-commit.yml

```diff
@@ -1,33 +1,22 @@
-name: Run pre-commit job
+name: lint
 
 on:
- push:
+  # trigger on pushes to any branch, but not main
+  push:
+    branches-ignore:
+      - master
+  # and also on PRs to main
+  pull_request:
     branches:
-    - main
- pull_request:
-    branches:
-    - main
+      - main
 
 jobs:
- pre-commit-job:
+  pre-commit-job:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        shell: bash -l {0}
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
         python-version: 3.9
```

Comment:
I think we should use a newer Pythonv version. My fork runs in Python 3.12. As discussed with @joeloskarsson, the repo should run on Python>=3.9. Any reason why 3.9 is pinned, that I am missing?

Reply:
I only used python 3.9 because the version here doesn't matter too much in that this is the python version used for doing the linting, i.e. no code execution only static analysis. I know there have been issues with python 3.12 and older versions of flake8 (I can't remember the issue right now, but I can change the ci setup to use python 3.12 and we can see :smile:).